app/src/main/java/com/imaginaryrhombus/proctimer/ui/timer/TimerFragment.kt

```diff
@@ -49,23 +58,35 @@ class TimerFragment : Fragment() {
                     val ringtone = RingtoneManager.getRingtone(this@run, ringtoneUri)
                     ringtone.play()
 
-                    val alertBuilder = AlertDialog.Builder(this@run)
-                        .setTitle(R.string.timer_end_dialog_text)
-                        .setPositiveButton(R.string.button_timer_end_dialog_next_button) { _, _ ->
+                    if (binding.continueBox?.isChecked == true) {
+                        viewModel.nextTimer()
+                        viewModel.startTick()
+                        val timerEndNotification = TimerEndNotification(requireContext())
+                        val dateFormat = SimpleDateFormat("HH:mm:ss", Locale.getDefault())
+                        val endTimeString = dateFormat.format(Date())
+                        timerEndNotification.open(getString(R.string.notify_ended).format(endTimeString))
+                        Handler(mainLooper).postDelayed({
                             ringtone.stop()
-                            viewModel.nextTimer()
-                        }
-                        .setNeutralButton(
-                            R.string.button_timer_end_dialog_next_start_button
-                        ) { _, _ ->
-                            ringtone.stop()
-                            viewModel.nextTimer()
-                            viewModel.startTick()
-                        }
-
-                    val dialog = alertBuilder.create()
-                    dialog.setCanceledOnTouchOutside(false)
-                    dialog.show()
+                        }, 3000)
+                    } else {
+                        val alertBuilder = AlertDialog.Builder(this@run)
+                            .setTitle(R.string.timer_end_dialog_text)
+                            .setPositiveButton(R.string.button_timer_end_dialog_next_button) { _, _ ->
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/fizxive/ProcTimer/blob/b31a11cf7f617ce8ac48044c6c809fd38815309c/app/src/main/java/com/imaginaryrhombus/proctimer/ui/timer/TimerFragment.kt#L74"></span>Exceeded max line length (100) (cannot be auto-corrected)</td>
    </tr>
  </tbody>
</table>